Modules/Quest/AvailableQuests.lua

```diff
@@ -0,0 +1,349 @@
+            local childQuests = QuestieDB.QueryQuestSingle(questId, "childQuests")
+            if childQuests then
+                for _, childQuestId in pairs(childQuests) do
+                    if (not completedQuests[childQuestId]) and (not currentQuestlog[childQuestId]) then
+                        local childQuestExclusiveTo = QuestieDB.QueryQuestSingle(childQuestId, "exclusiveTo")
+                        local blockedByExclusiveTo = false
+                        for _, exclusiveToQuestId in pairs(childQuestExclusiveTo or {}) do
+                            if QuestiePlayer.currentQuestlog[exclusiveToQuestId] or Questie.db.char.complete[exclusiveToQuestId] then
+                                blockedByExclusiveTo = true
+                                break
+                            end
+                        end
+                        if (not blockedByExclusiveTo) then
+                            QuestieDB.activeChildQuests[childQuestId] = true
+                            availableQuests[childQuestId] = true
+                            -- Draw them right away and skip all other irrelevant checks
+                            _DrawAvailableQuest(childQuestId)
+                        end
+                    end
+                end
+            end
+            if QuestieDB.IsComplete(questId) ~= -1 then -- The quest in the quest log is not failed, so we don't show it as available
+                return
+            end
+        end
+
+        if (
+            ((not showRepeatableQuests) and QuestieDB.IsRepeatable(questId)) or   -- Don't show repeatable quests if option is disabled
+            ((not showPvPQuests) and QuestieDB.IsPvPQuest(questId)) or            -- Don't show PvP quests if option is disabled
+            ((not showDungeonQuests) and QuestieDB.IsDungeonQuest(questId)) or    -- Don't show dungeon quests if option is disabled
+            ((not showRaidQuests) and QuestieDB.IsRaidQuest(questId)) or          -- Don't show raid quests if option is disabled
+            ((not showAQWarEffortQuests) and aqWarEffortQuests[questId]) or       -- Don't show AQ War Effort quests if the option disabled
+            (Questie.IsClassic and ((not currentIsleOfQuelDanasQuests) or currentIsleOfQuelDanasQuests[questId])) or      -- Don't show Isle quests for Classic
+            ((not showSoDRunes) and Questie.IsSoD and QuestieDB.IsSoDRuneQuest(questId)) -- Don't show SoD Rune quests with the option disabled
+        ) then
+            return
+        end
+
+        if (
+            (not QuestieDB.IsLevelRequirementsFulfilled(questId, minLevel, maxLevel, playerLevel)) or
+            (not QuestieDB.IsDoable(questId, debugEnabled))
+        ) then
+            --If the quests are not within level range we want to unload them
+            --(This is for when people level up or change settings etc)
+            -- TODO: Is still necessary?
+            QuestieMap:UnloadQuestFrames(questId)
```

Comment:
Could this added to the `if` below? Not sure when we are adding frames, so not sure when we need to unload them.

Technically the above should be true, because we are talking about available quests and these are tracked in `availableQuests`.